From ad8dd1669d202b4ab20562f0efc35af4f040f9d4 Mon Sep 17 00:00:00 2001
From: Johannes Thoma <johannes@johannesthoma.com>
Date: Sat, 9 Feb 2019 22:03:00 +0100
Subject: [PATCH 09/16] Use WinDRBD specific SOCK_LISTEN socket type.

For listening socket. Reason is that in Windows (before Windows 10
1703) wsk distiguishes between connection and listening sockets.
There is no such thing as a stream socket which can do both.
---
 drbd/drbd_transport_tcp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drbd/drbd_transport_tcp.c b/drbd/drbd_transport_tcp.c
index d91d00a..4c0378c 100644
--- a/drbd/drbd_transport_tcp.c
+++ b/drbd/drbd_transport_tcp.c
@@ -818,7 +818,7 @@ static int dtt_init_listener(struct drbd_transport *transport,
 
 	my_addr = *(struct sockaddr_storage *)addr;
 
-	err = sock_create_kern(&init_net, my_addr.ss_family, SOCK_STREAM, IPPROTO_TCP, &s_listen);
+	err = sock_create_kern(&init_net, my_addr.ss_family, SOCK_LISTEN, IPPROTO_TCP, &s_listen);
 	if (err) {
 		s_listen = NULL;
 		what = "sock_create_kern";
-- 
2.17.0

